topics/icrc_57_get_session_delegation.md

```diff
@@ -0,0 +1,160 @@
+    * [Example RPC Request](#example-rpc-request)
+    * [Response](#response)
+    * [Example RPC Response](#example-rpc-response)
+  * [Message Processing](#message-processing)
+  * [Errors](#errors)
+<!-- TOC -->
+
+## Summary
+
+When a relying party wants to authenticate as a user, it uses
+a [session key](https://internetcomputer.org/docs/current/references/ic-interface-spec/#ecdsa), and
+below `icrc57_get_session_delegation` method to obtain a delegation chain that allows the session key to sign for the
+user's session identity. The obtained delegation chain is scoped per relying party, resulting in a different identity
+for each relying party.
+
+Compared to a global identity ([ICRC-34](./icrc_34_get_global_delegation.md)), a session identity has no requirements
+and restrictions.
+
+- The relying party does not require to be trusted by canisters, so any canister can be called on behalf of the user.
+- There's no canister target restriction, making it compatible with multi canister architectures like e.g. OpenChat.
+- It is scoped per relying party, resulting in anonymity by default for the user between relying parties.
+
+## Method
+
+**Name and Scope:** `icrc57_get_session_delegation`
+
+**Prerequisite:** Active session with granted permission scope `icrc57_get_session_delegation` or `*`.
+
+## Scope (according to the [ICRC-25 standard](./icrc_25_signer_interaction_standard.md))
+
+**Scope:** `icrc57_get_session_delegation`
+
+**Optional Properties:**
+
+- `targets` (`text` array): A list of target canister ids (textual representation) the scope is restricted to.
+  If the list is not present, the scope applies to all canisters (i.e. the permission is not restricted).
+
+### Example RPC Request Permission
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc25_request_permissions",
+  "params": {
+    "scopes": [
+      {
+        "method": "icrc57_get_session_delegation",
+        "targets": [
+          "xhy27-fqaaa-aaaao-a2hlq-cai"
+        ]
+      }
+    ]
+  }
+}
+```
+
+## Request
+
+**`publicKey` (`blob`):** Public key that receives the global delegation as described in
+the [IC interface specification, signatures section](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures).  
+**`targets` (`text` array):** (Optional) A list of target canister ids (textual representation) the delegation is
+restricted to.
+If the list is not present, the delegation applies to all canisters (i.e. the delegation is not restricted).  
+**`maxTimeToLive` (`text` array):** (Optional) Expiration of the delegation in nanoseconds, signer can still choose to
+return a delegation with a shorter expiration.
+
+### Example RPC Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc57_get_session_delegation",
+  "params": {
+    "publicKey": "MDwwDAYKKwYBBAGDuEMBAgMsAAoAAAAAAGAAJwEB9YN/ErQ8yN+14qewhrU0Hm2rZZ77SrydLsSMRYHoNxM=",
+    "targets": [
+      "xhy27-fqaaa-aaaao-a2hlq-cai"
```

Comment:
The example is now mismatched: when requesting a target restriction, that target restriction should be honored by the returned delegation.
But because session isolation in this case is not done via targets, shouldn't we remove this parameter from the request all together?

Reply:
I've removed it, if e.g. a relying party want's to restrict it for security reasons, it could always extend the delegation chain itself.